invenio_rdm_records/marshmallow/json.py

```diff
@@ -12,53 +12,125 @@
 
 import arrow
 from flask_babelex import lazy_gettext as _
-from invenio_records_rest.schemas import Nested, StrictKeysMixin
+from invenio_records_rest.schemas import Nested
 from invenio_records_rest.schemas.fields import DateString, \
     PersistentIdentifier, SanitizedUnicode
+from invenio_rest.serializer import BaseSchema
 from marshmallow import ValidationError, fields, pre_load, validate, \
     validates, validates_schema
 
 from ..models import ObjectType
-from .utils import validate_iso639_3
+from .utils import api_link_for, validate_iso639_3
 
 
-class AccessSchemaV1(StrictKeysMixin):
-    """Access schema."""
+class AccessConditionSchemaV1(BaseSchema):
+    """Access condition."""
 
-    # TODO revist acccording to
-    # https://github.com/inveniosoftware/invenio-rdm-records/issues/20
-    metadata_restricted = fields.Bool(required=True)
-    files_restricted = fields.Bool(required=True)
+    condition = SanitizedUnicode()
+    link_validity = SanitizedUnicode()
 
 
-class PersonIdsSchemaV1(StrictKeysMixin):
-    """Ids schema."""
+class CommunitySchemaV1(BaseSchema):
+    """Communities to which the record belongs to."""
 
-    source = SanitizedUnicode()
-    value = SanitizedUnicode()
+    primary = SanitizedUnicode(required=True)
+    secondary = fields.List(SanitizedUnicode())
 
 
-class ContributorSchemaV1(StrictKeysMixin):
+class IdentifierSchemaV1(BaseSchema):
+    """Extra/Alternate identifiers of the record."""
+
+    identifier = SanitizedUnicode(required=True)
+    scheme = SanitizedUnicode(required=True)
+
+
+class AffiliationSchemaV1(BaseSchema):
+    """Affiliation of a creator/contributor."""
+
+    name = SanitizedUnicode(required=True)
+    identifier = SanitizedUnicode(required=True)
+    scheme = SanitizedUnicode(required=True)
+
+
+class CreatorSchemaV1(BaseSchema):
+    """Creator schema."""
+
+    NAMES = [
+        "Organizational",
+        "Personal"
+    ]
+
+    name = SanitizedUnicode(required=True)
+    type = SanitizedUnicode(required=True, validate=validate.OneOf(
```

Comment:
You are right, there is a difference in performance. I run some tests, first iteration on loading/dumping 100 times a string, second iteration 100000 times. Time difference seems to be linear (makes sense):

1 load/dump:

``` console
Sanitized: 0.0006556510925292969 seconds
Marshmallow Str: 0.0002028942108154297 seconds
```

1000 load/dumps:

``` console
Sanitized:  0.18001985549926758 seconds
Marshmallow Str: 0.09990668296813965 seconds
```

100000 load/dumps:

``` console
Sanitized: 18.859431743621826 seconds
Marshmallow Str: 10.20322585105896 seconds
```

However Im still not convince of leaving strings not sanitized. It is true that for now they are enums so there is no room for errors. But in the future if we open to customization this CVs, it might be a possible problem. Opening https://github.com/inveniosoftware/invenio-rdm-records/issues/53 to investigate in the future.

Reply:
Nice. Yes, `Str` only makes sense for cases where we don't have customized string that may contain unicode characters.